Namespace Handling Enhancements

kellyredding opened the pull request:
## adding in namespace equality logic

Two namespaces with the same named values should be equal to each
other.  I chose to accomplish this by just comparing their hash
representations.
## equality testing for class instance proxies

Two class instance proxies should be equal if their proxied namespaces
are in turn equal.  Don't want to do this for the Module or Class level
proxying b/c that involve overriding the '==' method for `Class` and
`Module` - no good can come of that.
## better inspect method for proxies

This method essentially has the same logic as `Namespace#inspect`,
however it uses the proxy's class and object id info.

kellyredding committed:
adding in namespace equality logic

Two namespaces with the same named values should be equal to each
other.  I chose to accomplish this by just comparing their hash
representations.

kellyredding committed:
equality testing for class instance proxies

Two class instance proxies should be equal if their proxied namespaces
are in turn equal.  Don't want to do this for the Module or Class level
proxying b/c that involve overriding the '==' method for `Class` and
`Module` - no good can come of that.

kellyredding committed:
better inspect method for proxies

This method essentially has the same logic as `Namespace#inspect`,
however it uses the proxy's class and object id info.

kellyredding commented:
@jcredding - look over this when you get a moment.  I added some namespace and proxy handling methods.  Basically equality checking for namespaces and proxies where appropriate and a better `inspect` method for proxies.

For namespace equality checking I just compared their `to_hash` representations - any problem with that?

Thanks man.

jcredding commented:
@kellyredding - cool stuff, I like having inspects defined a lot. Curiously, what happens if you compare a proxy/namespace with a non proxy/namespace:

``` ruby
class Something
  include NsOptions::Proxy
end
Something.new == 1
```

Not sure if Ruby does magic for you or if you have to worry about this case, Just curious. You have more changes coming or you want me to roll a new version?

kellyredding committed:
better equality testing

Before the equality testing required both things to be `Namespace`
objects or it would raise an exception.  No good.  Added an inequality
test for good measure.

kellyredding commented:
@jcredding - boom.  (please merge)

jcredding commented:
@kellyredding - merged. New version?

kellyredding commented:
@jcredding - not needed just yet so I'll wait for now.  I'll request a new
version when needed.  Thanks.

On Wed, May 23, 2012 at 1:58 PM, Collin Redding <
reply@reply.github.com

> wrote:
> 
> @kellyredding - merged. New version?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/teaminsight/ns-options/pull/47#issuecomment-5881567

kellyredding commented:
@jcredding - i can haz new version bump?

jcredding commented:
Done and done, v0.4.1 is out.
